test/plugins/pipelines.test.js

```diff
@@ -230,6 +253,82 @@ describe('pipeline plugin test', () => {
         });
     });
 
+    describe('GET /pipelines/{id}/secrets', () => {
+        const pipelineId = 'd398fb192747c9a0124e9e5b4e6e8e841cf8c71c';
+        const username = 'myself';
+        const scmUrl = 'git@github.com:screwdriver-cd/data-model.git#master';
+        let options;
+        let pipelineMock;
+        let userMock;
+
+        beforeEach(() => {
+            options = {
+                method: 'GET',
+                url: `/pipelines/${pipelineId}/secrets`,
+                credentials: {
+                    username,
+                    scope: ['user']
+                }
+            };
+            pipelineMock = getPipelineMocks(testPipeline);
+            pipelineMock.secrets = getSecretsMocks(testSecrets);
+            pipelineFactoryMock.get.resolves(pipelineMock);
+
+            userMock = getUserMock({ username });
+            userMock.getPermissions.withArgs(scmUrl).resolves({ push: true });
+            userFactoryMock.get.withArgs({ username }).resolves(userMock);
+        });
+
+        it('returns 404 for updating a pipeline that does not exist', (done) => {
+            pipelineFactoryMock.get.resolves(null);
+
+            server.inject(options, (reply) => {
+                assert.equal(reply.statusCode, 404);
+                done();
+            });
+        });
+
+        it('returns 403 when the user does not have push permissions', (done) => {
+            userMock.getPermissions.withArgs(scmUrl).resolves({ push: false });
+
+            server.inject(options, (reply) => {
+                assert.equal(reply.statusCode, 403);
+                done();
```

Comment:
It's fine for now, but `server.inject` returns a promise.  So all these tests can be written like:

```
return server.inject(options).then((reply) => {
...
```